api/proto/teleport/mfa/v1/mfa_service.proto

```diff
@@ -0,0 +1,198 @@
+  string name = 1;
+  // mfa_challenge contains the MFA challenge that the user must respond to.
+  AuthenticateChallenge mfa_challenge = 2;
+}
+
+// ValidateChallengeRequest is the request message for ValidateChallenge.
+message ValidateChallengeRequest {
+  // name is the resource name for the issued challenge. This must match the 'name' returned in CreateChallengeResponse
+  // to tie the validation to the correct challenge.
+  string name = 1;
+  // mfa_response contains the MFA challenge response provided by the user.
+  AuthenticateResponse mfa_response = 2;
+}
+
+// ValidateChallengeResponse is the response message for ValidateChallenge.
+message ValidateChallengeResponse {}
+
+// ReplicateValidatedMFAChallengeRequest is the request message for ReplicateValidatedMFAChallenge.
+message ReplicateValidatedMFAChallengeRequest {
+  // name is the resource name for the issued challenge. This must match the 'name' returned in CreateChallengeResponse
+  // to tie the upsert to the correct challenge.
+  string name = 1;
+  // payload is a value that uniquely identifies the user's session. This should match the payload in
+  // CreateChallengeRequest.
+  SessionIdentifyingPayload payload = 2;
+  // device contains information about the user's MFA device used to authenticate.
+  types.MFADevice device = 3;
+  // source_cluster is the name of the cluster where the validated challenge originated.
+  string source_cluster = 4;
+  // target_cluster is the name of the destination cluster where the validated challenge should be replicated to.
+  string target_cluster = 5;
+}
+
+// ReplicateValidatedMFAChallengeResponse is the response message for ReplicateValidatedMFAChallenge.
+message ReplicateValidatedMFAChallengeResponse {
+  // replicated_challenge is the validated MFA challenge that was replicated.
+  ValidatedMFAChallenge replicated_challenge = 1;
+}
+
+// VerifyValidatedMFAChallengeRequest is the request message for VerifyValidatedMFAChallenge.
+message VerifyValidatedMFAChallengeRequest {
+  // name is the resource name for the issued challenge. This must match the 'name' returned in CreateChallengeResponse
+  // to tie the retrieval to the correct challenge.
+  string name = 1;
+  // payload is a value that uniquely identifies the user's session. The client calling VerifyValidatedMFAChallenge MUST
+  // independently compute this value from session state. The server will verify it matches the payload supplied in
+  // CreateChallengeRequest to ensure the challenge is tied to the correct session.
+  SessionIdentifyingPayload payload = 2;
+  // source_cluster is the name of the cluster where the validated challenge originated.
+  string source_cluster = 3;
+}
+
+// VerifyValidatedMFAChallengeResponse is the response message for VerifyValidatedMFAChallenge.
+message VerifyValidatedMFAChallengeResponse {
+  // device contains information about the user's MFA device used to authenticate.
+  types.MFADevice device = 1;
```

Comment:
Friendly reminder to be careful when returning devices so we don't leak sensitive information back to the caller.